.pylintrc

```diff
@@ -0,0 +1,503 @@
+[MASTER]
```

Comment:
Do we use any custom pylint functionality?

Reply:
Yes, you need to [whitelist the `pycompwa.ui` module](https://github.com/ComPWA/pycompwa/pull/77/files/494c0f1598387b3d2cc926314f3cfbeecc4dadfd#diff-d6d3d73911fd3ad6f39e4027f763b307R6). In addition, pylint can be quite rigorous, so you want to add ignores over time.

Reply:
Just have a bit trouble with this large config, which mainly does standard pylint things. I'm a bit torn on what is best practice here

Reply:
I'm also not sure, the config for `flake8` is nicely handled under the `tox.ini` file. It has a similar situation, just that the default options were not added there. Perhaps that's the better approach: only add the options that were modified and add more as soon as there's a need for it.

Reply:
That would be definitely be more transparent.
I guess other repos just use a specific version of pylint with the default rules, and bump up the pylint version from time to time, to ensure no code style changes during normal PRs.

Reply:
Yes this is a bit of a problem for this pre-commit hook tbh. As you can see in [the new `.pre-commit-config.yaml`](https://github.com/ComPWA/pycompwa/blob/494c0f1598387b3d2cc926314f3cfbeecc4dadfd/.pre-commit-config.yaml), the other hooks like the one for `flake8` have well-defined versions, but the problem with `pylint` is that it does small runtime checks as well (hence whitelisting `pycompwa.ui`), which means that it needs to know the `pycompwa` installation in your virtual environment. That's why the hook for `pylint` is listed as a 'local' hook. But it also means the version is whatever you have installed locally.

A solution could be defining a `pylint` version in the [`requirements_dev.txt`](https://github.com/ComPWA/pycompwa/blob/494c0f1598387b3d2cc926314f3cfbeecc4dadfd/requirements_dev.txt) file, because I assume that's how you install `pylint` in your virtual environment.

Reply:
Hmm, yeah. I think you want to avoid these problems that it runs locally but fails on the CI because some version is different. Are we doing any checks based on pylint etc in the CI? Usually other ppl define a production requirements file for this, to ensure everything will be working.

Reply:
https://github.com/ComPWA/pycompwa/pull/77/commits/1b7d3c5acf7ea00bc9b8a52347ed040cd1582120
Afaik Codacy does pylint checks, it uses pylint 2.4.4

Reply:
But note that a pre-commit hook is slightly different than CI: it only checks staged files (and is supposed to do only that)